### PR TITLE
UI changes for generating short-lived impersonation keys.

### PR DESCRIPTION
**Pull request by vadimberezniker**
The impersonation banner will now have a button that triggers generation of a short-lived impersonation API key.

**Related issues**: N/A

### CODE REVIEW

File: enterprise/app/root/root.tsx [hunk 1]
```diff
@@ -68,6 +73,80 @@ capabilities.register("BuildBuddy Enterprise", true, [
...
+      <div className="impersonation-toolbar">
+        <AlertCircle className="icon black" />
+        <span>
+          Authenticated as a member of <b>{this.props.user.selectedGroupName()}</b> ({this.props.user.selectedGroup?.id}
+          ). Proceed with caution.
+        </span>
+        <OutlinedButton
+          onClick={this.handleGenerateImpersonationAPIKeyClicked.bind(this)}
+          className="generate-api-key-button">
+          <span>{this.state.apiKey ? "Copy" : "Generate"} Temporary API Key</span>
```
Comment (bduffany) [inline enterprise/app/root/root.tsx:138]:
```suggestion
          <span>{this.state.apiKey ? "Copy" : "Generate"} temporary API key</span>
```

File: enterprise/app/root/root.css [hunk 2]
```diff
@@ -21,8 +21,15 @@ body {
   box-shadow: 0 -1px 1px rgba(0, 0, 0, 0.12) inset;
 }
 
-.impersonation-toolbar button {
+.impersonation-toolbar .exit-button {
   margin-left: auto;
+}
+
+.impersonation-toolbar .generate-api-key-button {
+  margin-left: 30px;
```
Comment (bduffany) [inline enterprise/app/root/root.css:29]: uber nit: spacing values are typically multiples of 8px

```suggestion
  margin-left: 32px;
```

**Review by bduffany — APPROVED** [review]

File: enterprise/app/root/root.tsx (hunk 1 above)
Comment (vadimberezniker) [reply enterprise/app/root/root.tsx]: done

File: enterprise/app/root/root.css (hunk 2 above)
Comment (vadimberezniker) [reply enterprise/app/root/root.css]: done

**Pull request merged**